test/e2e/framework/test_context.go

```diff
@@ -154,6 +156,9 @@ type CloudConfig struct {
 	MasterTag         string
 
 	Provider cloudprovider.Interface
+
+	UseKubemarkProvider bool
```

Comment:
Can we avoid this extra flag? I guess we can just use `ExternalKubeConfig` (if specified) to create `KubemarkProvider` when we know that `IsProvider("kubemark") == true`.